src/app/conversionTracking.service.ts

```diff
@@ -2,21 +2,30 @@ import {MatomoTracker} from "ngx-matomo";
 import {Injectable} from "@angular/core";
 import {Donation} from "./donation.model";
 import {Campaign} from "./campaign.model";
+import {CookiePreferenceService} from "./cookiePreference.service";
 
 @Injectable({
   providedIn: 'root',
 })
 export class ConversionTrackingService {
+  private marketingCookiesAllowed: boolean = false;
   constructor(
     private matomoTracker: MatomoTracker,
+    private cookiePreferenceService: CookiePreferenceService,
   ) {
+    this.cookiePreferenceService.userOptInToMarketingCookies().subscribe(() => this.marketingCookiesAllowed = true);
   }
 
+
   public convert(donation: Donation, campaign: Campaign) {
     this.trackConversionWithMatomo(donation, campaign);
   }
 
   private trackConversionWithMatomo(donation: Donation, campaign: Campaign) {
+    if (! this.marketingCookiesAllowed) {
+      return;
```

Comment:
Although we're not using the matomo eccomerce tracking feature in this case we can still count donations via matomo using the `thank_you_fully_loaded` event that we send from https://github.com/thebiggive/donate-frontend/blob/02727e2a1c9c8231c7edd5001d18d601e017348d/src/app/donation-complete/donation-complete.component.ts#L230